sources/views/contacts.js

```diff
@@ -52,12 +59,36 @@ export default class ContactsView extends JetView {
 	}
 	init() {
 		this.$$("list").sync(contacts);
-	}
-	urlChange() {
+		this.contactForm = this.ui(ContactFormView);
+
+		this.on(this.app, "onCallContactForm", way => {
+			this.show(`contactForm?way=${way}`);
+			this.$$("list").disable();
+		});
+		this.on(this.app, "onAfterDelete_contactDetailed", () => {
+			this.$$("list").select(contacts.getFirstId());
+		});
+		this.on(this.app, "onClickCancel_contactForm", way => {
+			this.$$("list").enable();
+			this.show("contactDetailed");
+			if (way == "edit") {
+				this.$$("list").select(this.getParam("id", true));
+			} else if (way == "add") {
+				this.setParam("id", contacts.getFirstId(), true);
+				this.$$("list").select(contacts.getFirstId());
```

Comment:
[Согласно ТЗ](https://hackmd.io/s/rk_RV6fez#Scenarios-general), селектиться после Add должен добавленный контакт: 

> _I want to click on ‘Add’ so that the new contact will be added to the list and the form will be switched to a page with contact’s details._

Так как реальную ID добавленный айтем получает после ответа сервера, можно использовать событие [onIdChange](https://docs.webix.com/api__datastore_onidchange_event.html) для датастора коллекции.